integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -268,6 +267,11 @@ private[sql] case class CarbonProjectForUpdateCommand(
         // When the table has too many segemnts, it will take a long time.
         // So moving it to the end and it is outside of locking.
         CarbonUpdateUtil.cleanStaleDeltaFiles(carbonTable, fileTimestamp)
+
+        // Delete The New Inserted Segment
+        CarbonLoaderUtil.updateTableStatusInCaseOfFailure(
```

Comment:
1. better to get loadName from the insertInto command
2. update and insert should update tablestatus once at the end.
  if insert success but update not finish or failed, in some time another concurrent query will give a wrong result.
  it shouldn't break ACID rule.
3. refresh index cache for deleted segment only, it should not contain updated segment.

Reply:
I have modified code according to your suggestion